File: story3.5/ConcertSeatManager.java
/**
 * Filename : ConsertSeatManager.java
 * Description :  determine the first customer who requests a seat that has already been
requested by someone else. 
 * Author : Nancy Jain
 */

import java.util.Set;
import java.util.HashSet;
public class ConcertSeatManager {
    public static String DuplicateDetection(String[][] requests){
        Set<String> seats = new HashSet<>();
        for(String[] req : requests){
            if(!seats.add(req[1])){
                return req[0];
            }
        }
        return "-1";
    }
}

File: week3.5_story/ConcertSeatManager.cpp
/**
 * @filename   : ConcertSeatManager.cpp
 * @description: Detect the first customer who requests a seat that has already been assigned.
 * @author     : Ehtesham Nawaz
 */

#include <bits/stdc++.h>
using namespace std;



string findFirstDuplicateSeat(const vector<pair<string, int>>& requests) {
  
   unordered_map<int,string>assignedSeats;

            for(auto [customerId,seatNumber]:requests){
                if(assignedSeats.find(seatNumber)!=assignedSeats.end()){
                    return customerId;
                }else{
                    assignedSeats[seatNumber]=customerId;

                }
            }

            return "-1";
}

int main() {
    int numberOfRequests;
    cin >> numberOfRequests;   

    vector<pair<string, int>> requests;
    for (int i = 0; i < numberOfRequests; i++) {
        string customerId;
        int seatNumber;
        cin >> customerId >> seatNumber;
        requests.push_back({customerId, seatNumber});
    }

    cout << findFirstDuplicateSeat(requests) << endl;

    return 0;
}
